File: 20221012/모범답안/20221012_10/CountBingo_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.hard.CountBingo.CountBingo import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        board = [[True, True, True], [True, True, True], [False, False, True]]

        # when
        result = solution(board)

        # then
        self.assertEqual(3, result)

    def test_2(self):
        # given
        board = [[True, True, True], [True, True, True], [True, True, True]]

        # when
        result = solution(board)

        # then
        self.assertEqual(6, result)

    def test_3(self):
        # given
        board = [[True]]

        # when
        result = solution(board)

        # then
        self.assertEqual(2, result)

    def test_4(self):
        # given
        board = [[False]]

        # when
        result = solution(board)

        # then
        self.assertEqual(0, result)

    def test_5(self):
        # given
        board = [[True, False, True], [False, True, False], [True, False, True]]

        # when
        result = solution(board)

        # then
        self.assertEqual(0, result)

    def test_6(self):
        # given
        board = [[False, True, False], [True, False, True], [False, True, False]]

        # when
        result = solution(board)

        # then
        self.assertEqual(0, result)


if __name__ == '__main__':
    unittest.main()

File: 20221012/모범답안/20221012_10/CountBingo_test.js
import assert from 'assert'
import solution from './CountBingo.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const board = [[true, true, true], [true, true, true], [false, false, true]]

    // when
    const result = solution(board)

    // then
    assert.deepStrictEqual(3, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const board = [[true, true, true], [true, true, true], [true, true, true]]

    // when
    const result = solution(board)

    // then
    assert.deepStrictEqual(6, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const board = [[true]]

    // when
    const result = solution(board)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const board = [[false]]

    // when
    const result = solution(board)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const board = [[true, false, true], [false, true, false], [true, false, true]]

    // when
    const result = solution(board)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const board = [[false, true, false], [true, false, true], [false, true, false]]

    // when
    const result = solution(board)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });
})
